### PR TITLE
[WIP] Feature/treeshaking improvements

### PR DESCRIPTION
**Pull request by tysonwolker**
This change is [Reviewable](https://reviewable.io/reviews/qantasairways/runway/183)

### CODE REVIEW

File: babel.config.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+          : 'commonjs'
+      }
+    ]
+  ];
+}
+
+const productionPlugins = [
+  'babel-plugin-transform-react-constant-elements',
+  'babel-plugin-transform-dev-warning',
+  ['babel-plugin-react-remove-properties', { properties: ['data-test-id'] }],
```
Comment (jamesb3ll) [inline babel.config.js]: I think it's normally `data-testid` and that's what can be used with React Testing Library with `getByTestId`:
https://testing-library.com/docs/dom-testing-library/api-queries#bytestid

File: src/components/SvgIcon/SvgIcon.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { css } from '@emotion/core';
+
+export function styles({ width, height }) {
+  return css({
+    userSelect: 'none',
+    width,
```
Comment (rachmulvey) [inline src/components/SvgIcon/SvgIcon.js]: I think you probably need to append a unit like `px` to width and height if we are doing it through css

File: .circleci/config.yml [hunk 3]
```diff
@@ -66,6 +66,9 @@ jobs:
       - run:
           name: Authenticate with registry
           command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/.npmrc
+      - run:
+        name: Change directory to distribution folder to publish
+        command: cd lib
```
Comment (rachmulvey) [inline .circleci/config.yml]: this might not work without indenting `name` and `command`

File: rollup.config.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      nodeResolve(),
+      babel(babelOptions),
+      commonjs(commonjsOptions),
+      nodeGlobals(), // Wait for https://github.com/cssinjs/jss/pull/893
+      replace({ 'process.env.NODE_ENV': JSON.stringify('production') }),
+      sizeSnapshot({ snapshotPath: 'size-snapshot.json' }),
+      terser()
+    ]
+  }
+];
```
Comment (rachmulvey) [inline rollup.config.js]: this is sooo much better 😍

File: .circleci/config.yml (hunk 3 above)
Comment (tysonwolker) [reply .circleci/config.yml]: Thanks made me realise i need to copy semantic release too 😬 and exclude node_modules

File: rollup.config.js (hunk 4 above)
Comment (tysonwolker) [reply rollup.config.js]: I might add watch commands, depending on how i feel

- [ ] Watch commands?

File: babel.config.js (hunk 1 above)
Comment (tysonwolker) [reply babel.config.js]: Done

File: babel.config.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
+let defaultPresets;
+
+if (process.env.BABEL_ENV === 'es') {
+  defaultPresets = [];
```
Comment (jamesb3ll) [inline babel.config.js]: For ES should we maybe include `@babel/preset-modules` which targets modern browsers and also works around bugs and inconsistencies in modern JavaScript engines
https://github.com/babel/preset-modules

**Comment by jamesb3ll** [comment]
LGTM = Lets Get This Merged!!! 😄

**Pull request closed without merging**